lib/parse_cmds.js

```diff
@@ -0,0 +1,543 @@
+// need to set all these locally in function scope
```

Comment:
There won't be any race condition, it's single threaded, and all these operations are blocking or fire and forget.

Anyway, closing my pull request, Won't have time to do this.
(Most of these changes are just imported from my own project.)